fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java

```diff
@@ -937,6 +950,20 @@ protected CompletableFuture<Void> iterateAllRanges(List<Object> additionalLogMes
                 getRunner().getExecutor());
     }
 
+    public CompletableFuture<Void> mergeIndexes() {
+        return AsyncUtil.whenAll(luceneIndexes.stream()
```

Comment:
Here we know about Lucene indexes as supporting merge, even though we pretended elsewhere that it's general.

I think a better approach might be to have the indexer capable of doing throttle-able `IndexOperations` and making non-inline Lucene merge one of those.

Reply:
Perhaps the item missing here is the ability to determine whether there is any reason to do this at all. There's no way for the indexer to determine which, if any of the indexes would benefit from calling `mergeIndexes`.
That being said, if they are all no-op, I doubt that would have any noticeable impact on any of the indexing performance.